script/dash-lint.sh

```diff
@@ -1,18 +1,25 @@
 #!/usr/bin/env bash
 
 TOOL_NAME=$1
-INCORRECT_NAME=$(echo "$TOOL_NAME" | tr -d -)
-
-CURRENT_FILE=$(basename "${BASH_SOURCE}")
 ROOT_DIR=$(git rev-parse --show-toplevel)
-BAD_NAMES=$(grep -ril "$INCORRECT_NAME" "$ROOT_DIR" --exclude "$CURRENT_FILE" --exclude-dir=".git")
 
-if [ -n "$BAD_NAMES" ]; then
-    echo "$TOOL_NAME has a dash in it, and inconsistencies in the codebase can cause major problems, please correct the spelling." >&2
-    printf "The following files contain the incorrect string '%s':\n\n" "$INCORRECT_NAME" >&2
-    echo "$BAD_NAMES" >&2
-    exit 1;
-else
-    echo "No files contain the name '$INCORRECT_NAME'."
-    exit 0;
-fi
+function bad_delimiter_check() {
+    DELIMITER=$1
+    # shellcheck disable=SC2116 # Without echo, WRONG_NAME will be empty
+    WRONG_NAME=$(echo "${TOOL_NAME//-/$DELIMITER}")
+    echo "Checking for $WRONG_NAME in the codebase..."
+    FAILURES=$(grep -ril "$WRONG_NAME" "$ROOT_DIR")
+
+    if [ -n "$FAILURES" ]; then
+        echo "$TOOL_NAME has a dash in it, and inconsistencies in the codebase can cause major problems, please correct the spelling." >&2
```

Comment:
Perhaps something like if now covers more characters:
```suggestion
        echo "$TOOL_NAME has a "'$DELIMITER'" in it, and inconsistencies in the codebase can cause major problems, please correct the spelling." >&2
```

Reply:
Thanks! I've slightly modified your original suggestion and committed it!